src/account/CryptoTransferTransaction.ts

```diff
@@ -42,31 +43,33 @@ export class CryptoTransferTransaction extends TransactionBuilder {
         }
     }
 
-    addSender(accountId: AccountId, amount: number | BigNumber): this {
-        if (amount < 0) {
-            throw new Error('amount for addSender() must be nonnegative');
-        }
+    addSender(accountId: AccountId, amount: number | BigNumber | Hbar): this {
+        if (amount instanceof Hbar || amount instanceof BigNumber) {
+            if (amount.isNegative()) {
+                throw new Error('sending amount may not be negative');
+            }
 
-        return this.addTransfer(accountId, -amount);
-    }
+            return this.addTransfer(accountId, amount.negated());
```

Comment:
This is really cool: Typescript can resolve common fields/methods for sum types and project a sum type from that; the return type of this method call is `Hbar | BigNumber` which suits the signature of `addTransfer()`.